OpenSim/Simulation/SimbodyEngine/BallJoint.h

```diff
@@ -52,14 +52,36 @@ class OSIMSIMULATION_API BallJoint : public Joint {
     OpenSim_DECLARE_CONCRETE_OBJECT(BallJoint, Joint);
 
     /** Specify the Coordinates of the BallJoint */
-    CoordinateIndex rx{ constructCoordinate(Coordinate::MotionType::Rotational) };
-    CoordinateIndex ry{ constructCoordinate(Coordinate::MotionType::Rotational) };
-    CoordinateIndex rz{ constructCoordinate(Coordinate::MotionType::Rotational) };
+    CoordinateIndex rx{constructCoordinate(Coordinate::MotionType::Rotational)};
+    CoordinateIndex ry{constructCoordinate(Coordinate::MotionType::Rotational)};
+    CoordinateIndex rz{constructCoordinate(Coordinate::MotionType::Rotational)};
 
 public:
     /** Use Joint's constructors. @see Joint */
     using Joint::Joint;
 
+    using Joint::getCoordinate;
+
+    /** Indices of Coordinates; can be used with getCoordinate() for
+        convenience. */
+    enum class Coord {
+        Rotation1,
+        Rotation2,
+        Rotation3
+    };
+
+    /** Get the Coordinates associated with this Joint. */
+    const Coordinate& getCoordinate(Coord idx) const {
```

Comment:
A question: Do you think this method should be on the base `Joint` class (they could be virtual, and implemented by the subclasses). For example:

```
const auto& joint = model.getComponent<Joint>("/arm26/shoulder");
joint.getCoordinate(BallJoint::Coord::Rotation1);
```

I am undecided.

My gut is that if the methods are generic (e.g., they take `int`s), they belong in the base class. If they are in the subclasses, why not make them even more tailored to the specific subclasses? E.g., `BallJoint::getCoordinateR1()` or `getRotation1()`. I understand that you are using the `Coord` enum as an argument, which means that the method with the current signature could not be in the base class. But the signature could easily take an int, and one could pass a `Coord` for that int.